### PR TITLE
Updates to WinForms interop

### LINKED ISSUE #4951: WinFormsAvaloniaControlHost: Renderer is not started by default
Renderer is not started by default if Avalonia app is hosted in WinFormsAvaloniaControlHost. 
To reproduce see WindowsInteropTest/EmbedToWinFormsDemo.

As a workaround I can start it manually by using:
((TopLevel)content.GetVisualRoot()).Renderer.Start();

### LINKED ISSUE #4952: WpfAvaloniaHost: something is wrong with layout
Something is wrong with layout if we use WpfAvaloniaHost.
Check XAML:
https://github.com/JetBrains/Avalonia/blob/repro-interop-issues/samples/ControlCatalog/Pages/InteropIssuesPage.xaml

How this View looks like in Avalonia window/WinFormsAvaloniaControlHost (everything is OK):
image

And how this View looks like in WpfAvaloniaHost:
image

This issue is also valid for ListBox:
image

Repro is here (run WindowsInteropTest.csproj):
https://github.com/JetBrains/Avalonia/tree/repro-interop-issues

### LINKED ISSUE #4953: WpfAvaloniaHost/WinFormsAvaloniaControlHost: popups are not closed in some cases
Check how ContextMenu popup works in WpfAvaloniaHost/WinFormsAvaloniaControlHost:
repro3

And the same without WpfAvaloniaHost/WinFormsAvaloniaControlHost:
repro4

Repro is here (run WindowsInteropTest.csproj and ControlCatalog.NetCore.csproj):
https://github.com/JetBrains/Avalonia/tree/repro-interop-issues-popups

### LINKED ISSUE #5019: WpfAvaloniaHost: Input events are not handled
https://github.com/AvaloniaUI/Avalonia/blob/0f61b3cdd6d32f131041c68721f6216770378b9e/src/Windows/Avalonia.Win32.Interop/Wpf/WpfTopLevelImpl.cs#L210

Here and for other Input events `RawInputEventArgs.Handled` flag handling is missing.

### LINKED ISSUE #5865: WpfAvaloniaHost: WpfTopLevelImpl invalidates whole area on WpfTopLevelImpl.OnRender()
**Describe the bug**
Check out two gifs:
1. ControlCatalog.Desktop sample
2. WindowsInteropTest sample (hosted in WPF)

(Visualize dirty rects was enabled in DevTools)

**Expected behavior**
Dirty rects visualization will be the same on both scnearios.

1.
pure

2.
wpf_host

**Desktop:**

- OS: Windows
- Version: master

### LINKED ISSUE #8816: Attempt to close WPF window containing Avalonia control causes InvalidOperationException
**Describe the bug**

Whenever I try to close window from Avalonia control, exception is thrown. Problem is that `IInputRoot` doesn't have visual parent and therefore cannot convert point. Adding check for `_inputRoot.VisualParent is { }` seems to do the trick and window closes normally, but I'm not sure if this is good way to fix issue

https://github.com/AvaloniaUI/Avalonia/blob/97a18c8d675cb81e8bb1936f9570b30a5e101eba/src/Avalonia.Base/Input/PointerOverPreProcessor.cs#L31-L33

**To Reproduce**

1. Create WPF application
2. Add reference to Avalonia.Win32.Interoperability 
3. Create WPF window with Avalonia content
4. Open window with `Show`
5. Close window from Avalonia content by button click

Repository with example: https://github.com/jl0pd/AvaWindowFail

**Expected behavior**

Window closed normally

**Actual behavior**

Application throws following exception

```
InvalidOperationException: Visual is not connected to a PresentationSource
   at System.Windows.Media.Visual.PointFromScreen(Point point)
   at Avalonia.Win32.Interop.Wpf.WpfTopLevelImpl.Avalonia.Platform.ITopLevelImpl.PointToClient(PixelPoint point)
   at Avalonia.Controls.TopLevel.Avalonia.Rendering.IRenderRoot.PointToClient(PixelPoint p)
   at Avalonia.VisualExtensions.PointToClient(IVisual visual, PixelPoint point)
   at Avalonia.Input.PointerOverPreProcessor.OnNext(RawInputEventArgs value)
   at System.Reactive.Subjects.Subject`1.OnNext(T value)
   at Avalonia.Input.InputManager.ProcessInput(RawInputEventArgs e)
   at Avalonia.Controls.TopLevel.HandleInput(RawInputEventArgs e)
   at Avalonia.Win32.Interop.Wpf.WpfTopLevelImpl.MouseEvent(RawPointerEventType type, MouseEventArgs e)
   at Avalonia.Win32.Interop.Wpf.WpfTopLevelImpl.OnMouseLeave(MouseEventArgs e)
   at System.Windows.UIElement.OnMouseLeaveThunk(Object sender, MouseEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.ReverseInheritProperty.FirePropertyChangeInAncestry(DependencyObject element, Boolean oldValue, DeferredElementTreeState treeState, Action`2 originChangedAction)
   at System.Windows.ReverseInheritProperty.OnOriginValueChanged(DependencyObject oldOrigin, DependencyObject newOrigin, IList`1 otherOrigins, DeferredElementTreeState& oldTreeState, Action`2 originChangedAction)
   at System.Windows.Input.MouseDevice.ChangeMouseOver(IInputElement mouseOver, Int32 timestamp)
   at System.Windows.Input.MouseDevice.PreNotifyInput(Object sender, NotifyInputEventArgs e)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.MouseDevice.Synchronize()
   at System.Windows.Input.MouseDevice.ChangeMouseCapture(IInputElement mouseCapture, IMouseInputProvider providerCapture, CaptureMode captureMode, Int32 timestamp)
   at System.Windows.Input.MouseDevice.PreNotifyInput(Object sender, NotifyInputEventArgs e)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
```

**Desktop (please complete the following information):**

- OS: Windows 10 21h2
- Version 0.10.18, 11.0.0-preview1

### LINKED ISSUE #11454: Hosting Avalonia User Controls in Windows Forms on DotNet 7+
**Is your feature request related to a problem? Please describe.**

There is a need to gradually migrate some of our WinForms apps to Avalonia. To that end, we need to slowly, piece-by-piece replace WinForms user controls with Avalonia alternatives, eventually replacing the whole main window with one based on Avalonia. This approach requires us to be able to host Avalonia controls within WInForms Forms / UserControls.

So, I was trying to figure out a way to do it. The reverse  (hosting WinForms control inside of Avalonia app) seems pretty straight forward. Not so however with the other way around. I stumbled across `Avalonia.Win32.Interoperability` nuget package, however it was for an older Avalonia version and more crucially seemed to be targeting .Net Framework (as opposed to DotNet 7+). Researching a bit more I came across: [WinFormsAvaloniaControlHost](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Windows/Avalonia.Win32.Interop/WinForms/WinFormsAvaloniaControlHost.cs) from that same package.

So, I tried using it directly in my DotNet7 project. It was pretty easy to port. However there were 2 remaining issues:
- `private IntPtr WindowHandle => ((WindowImpl) _root?.PlatformImpl)?.Handle?.Handle ?? IntPtr.Zero;` could not be ported because apparently recently `WindowImpl` was made `internal`, nor was there a public interface on it that exposes the needed `IPlatformHandle Handle` propety.
- `EmbeddableControlRoot.FocusManager` was not there which I needed to port `if (_root.IsFocused) _root.FocusManager.ClearFocus();`, though this seemed less important.

I resolved the first issue with an ugly reflection hack, basically gaining access to `((WindowImpl) _root?.PlatformImpl)?.Handle?.Handle` through reflection, which for the purposes of testing sufficed.

So then I proceeded with a simple code to host my Avalonia control inside of WinForms:

```cs

  public Form1() {
    InitializeComponent();
    var avahost = new WinFormsAvaloniaControlHost(); // <-- Implemented in my project as I ported it from https://github.com/AvaloniaUI/Avalonia/blob/master/src/Windows/Avalonia.Win32.Interop/WinForms/WinFormsAvaloniaControlHost.cs
    panel1.Controls.Add(avahost);
    avahost.Dock = DockStyle.Fill;
    avahost.Content = new UserControl1();
  }
```

However once I ran my program, I would always be getting a pure white background in place where my Avalonia control should be.

**Describe the solution you'd like**
Basically I should just be able to host Avalonia control within any WinForms Form or User Control, so that we can migrate large WinForms app slowly over time, and this scenario is important not just for .Net Framework, but for modern DotNet (i.e. 7+)

- As an aside, for such scenarios it is especially important to be able to access Avalonia control's native handles (when they exist), such that there is no need to use Reflection hacks. These handles should be part of a public surface APIs.
- Also how should Avalonia be properly initialized for such a scenario (i.e. where the main program / app is WInForms). Do we just use `BuildAvaloniaApp().SetupWithoutStarting();` in our porogram main before we do `System.Windows.Forms.Application.Run(new Form1());`, or do we do something else?

**Describe alternatives you've considered**
There are really no feasible alternatives. Hosting WinForms inside Avalonia - doing it the other way around in other words is not really possible since the whole form would somehow need to be hosted inside of Avalonia. Not to mention there will still be a need to host Avalonia controls inside of WinForms container controls (until the entire UI was migrated to Avalonia).

**Additional context**
Using DotNet 7+ in VS 2022.

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

1. Renames Avalonia.Win32.Interop to Avalonia.Win32.Interoperability as that's how this project was named on the nuget.
2. Fixes CI configuration to build Avalonia.Win32.Interoperability project again. For reference, it wasn't published to nuget singe 11.0-preview1 (I assume because we dropped old MSBuild around there, and forgot to adjust config).
3. Adds "net6.0-windows" to the Avalonia.Win32.Interoperability. Still buildable on non-windows.
4. ~Replaces old flucky implementation of WpfAvaloniaHost with one based on HwndHost. Old implementation potentially could work better and deeper integrated, but didn't maintain it at all, leaving that in a bad state. HwndHost-based solution matches the behavior of WinForms Avalonia control and in works better in general.~
5. **WPF implementation was removed after internal discussion**. It might be returned at some point when we implement proper backend for WPF. Before that, if you need to host Avalonia control in WPF, and don't mind airspace problem, please use combination of HwndHost and EmbeddableControlRoot. See https://github.com/maxkatz6/AvaloniaHwndHostSample/blob/main/Avalonia.Win32.Interoperability/AvaloniaHwndHost.cs for reference.
6. Both controls do start render timer automatically now and handle correct control lifecycle events (handle created/destroyed) instead of inializing right away in ctor.
7. Some minor fixes in ControlCatalog.Desktop.

## Breaking changes

- Namespace change.
- WPF control was removed.

## Fixed issues

Fixes #11454
Fixes #4951
Fixes #5865
Fixes #8816
Fixes #5019
Fixes #4953 (possibly, but I couldn't repro even before)
Fixes #4952

### CODE REVIEW

File: dirs.proj [hunk 1]
```diff
@@ -14,11 +14,11 @@
     <ProjectReference Remove="samples/*.Android/*.csproj" />
     <ProjectReference Remove="samples/*.Browser/*.csproj" />
     <ProjectReference Remove="samples/*.Blazor/*.csproj" />
+    <ProjectReference Remove="samples/ControlCatalog.Desktop/*.*proj" />
   </ItemGroup>
-  <ItemGroup Condition="!$([MSBuild]::IsOsPlatform('Windows')) OR '$(MSBuildRuntimeType)' != 'Full'">
```
Comment (maxkatz6) [inline dirs.proj:18]: MSBuildRuntimeType is never "full" now, this project was never built on CI.

File: tests/Avalonia.UnitTests/InvariantCultureFixture.cs [hunk 2]
```diff
@@ -17,22 +17,13 @@ public class InvariantCultureFixture : IDisposable
 
         public InvariantCultureFixture()
         {
-#if NET461
-            _restore = Thread.CurrentThread.CurrentCulture;
-            Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;
-#else
```
Comment (maxkatz6) [inline tests/Avalonia.UnitTests/InvariantCultureFixture.cs:23]: This project is only targeting net standard, so this condition was never met.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035274-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by Mrxx99** [comment]
I don't think airspace is a huge problem with WpfAvaloniaHost in most cases, as the most likely overlap are popups or opened combobox contents which already work fine. So I don't understand why it will be removed. There are already blog posts out there that show how to use it those will now be useless and lead to frustration if trying to follow

**Comment by maxkatz6** [comment]
@Mrxx99 @robloo 

Yes, it shouldn't be a big problem for existing applications for a couple of reasons:
1. Old wpf host control was broken with lots of issues:
- forcing direct2d + sharpdx (=bad perf)
- forcing ui-thread renderer (=bad perf x2)
- wheel scrolling is just broken for some reason
- no adorners at all for some reason
- tab focus is broken for some reason
- embedding is not implemented (though, not a big deal)
- also https://github.com/AvaloniaUI/Avalonia/issues/4952
- and https://github.com/AvaloniaUI/Avalonia/issues/5019
- and https://github.com/AvaloniaUI/Avalonia/issues/8816
2. From the nuget statistics, this package was rarely used, one of the least downloaded packages we even have. And nobody even noticed that this package wasn't shipped to the nuget since 11.0-preview1
3. There is a simple drop in replacement based on HwndHost (see [my repo](https://github.com/maxkatz6/AvaloniaHwndHostSample/blob/main/Avalonia.Win32.Interoperability/AvaloniaHwndHost.cs)). Which I have recommended a while before.

XPF reasoning here related to the fact, that it is possible to implement a proper backend for WPF (not only XPF) without HwndHost in the future. While it's less possible for WinForms.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035850-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by grokys — APPROVED** [review]

**Pull request merged**